gateway/plugin/external.go

```diff
@@ -85,38 +85,30 @@ func (s ExternalServiceQuery) GetReplicas(serviceName string) (handlers.ServiceQ
 		}
 	}
 
+	minReplicas := uint64(handlers.DefaultMinReplicas)
 	maxReplicas := uint64(handlers.DefaultMaxReplicas)
-	minReplicas := uint64(1)
+	scalingFactor := uint64(handlers.DefaultScalingFactor)
 	availableReplicas := function.AvailableReplicas
 
 	if function.Labels != nil {
 		labels := *function.Labels
-		minScale := labels[handlers.MinScaleLabel]
-		maxScale := labels[handlers.MaxScaleLabel]
 
-		if len(minScale) > 0 {
-			labelValue, err := strconv.Atoi(minScale)
-			if err != nil {
-				log.Printf("Bad replica count: %s, should be uint", minScale)
-			} else {
-				minReplicas = uint64(labelValue)
-			}
-		}
+		minReplicas = extractLabelValue(labels[handlers.MinScaleLabel], minReplicas)
+		maxReplicas = extractLabelValue(labels[handlers.MaxScaleLabel], maxReplicas)
+		temp := extractLabelValue(labels[handlers.ScalingFactorLabel], scalingFactor)
```

Comment:
How about calling `temp` `scalingFactor`. temp is not a descriptive name?

Reply:
Sure `temp` is not descriptive. But using `scalingFactor ` would be not a good idea as the provided value could still be in an illegal range and in that case we would require a fallback to the default value, which we would have overridden. I will make the name more descriptive.